src/common/buffer.cc

```diff
@@ -1172,7 +1172,7 @@ using namespace ceph;
     if (p == ls->end())
       seek(off);
     unsigned left = len;
-    for (std::list<ptr>::const_iterator i = otherl._buffers.begin();
+    for (buffers_t::const_iterator i = otherl._buffers.begin();
```

Comment:
Why not just use auto for the type of an iterator returned by begin or whatnot?

Reply:
Sure, this gives a nice opportunity for a refactor. Switching to modern constructs. :-)